yt_dlp/YoutubeDL.py

```diff
@@ -3582,6 +3577,16 @@ def download_with_info_file(self, info_filename):
                 self.report_error(e)
         return self._download_retcode
 
+    def download_with_info_file(self, info_filename):
+        """Download using an info file."""
+        with contextlib.closing(fileinput.FileInput(
+                [info_filename], mode='r',
+                openhook=fileinput.hook_encoded('utf-8'))) as f:
+            # FileInput doesn't have a read method, we can't call json.load
+            infos = [self.sanitize_info(info, self.params.get('clean_infojson', True))
+                     for info in variadic(json.loads('\n'.join(f)))]
+        return self.download_with_info(infos)
```

Comment:
Sanitization should be inside `download_with_info`. Passing an unsanitized dict can lead to unexpected behavior